mitmproxy/tools/web/app.py

```diff
@@ -309,13 +309,17 @@ def get(self):
         self.set_header("Content-Disposition", "attachment; filename=flows")
         self.set_header("Content-Type", "application/octet-stream")
 
-        bio = BytesIO()
-        fw = io.FlowWriter(bio)
-        for f in self.view:
-            fw.add(f)
-
-        self.write(bio.getvalue())
-        bio.close()
+        try:
+            match = flowfilter.parse(self.request.arguments["filter"][0].decode())
+        except (KeyError, IndexError, ValueError):  # Key+Index: ["filter"][0], Value: parsing problem
```

Comment:
We should return an HTTP error if the filter is not valid (and add a test for this), so let's not just catch the ValueError here. :)

Reply:
Ok, I split the except up and threw a 400 (without reflecting the filter, as that would be reflected XSS)

And I ran the update command, which updated a HeaderSpec.tsx.map

Just to be sure:
The files app.css, app.js and vendor.css are also dirty but I shouldn't upload them as that's updated by some CI/CD?

Reply:
Yes. We include the compiled web assets in the repo so that Python devs don't need to deal with an entire Node toolchain. We manually update them before releases to avoid merge conflicts in the meantime.